Neos.Flow/Classes/Mvc/Routing/Router.php

```diff
@@ -267,9 +276,112 @@ protected function createRoutesFromConfiguration()
             }
             $this->routes[] = $route;
         }
+
         $this->routesCreated = true;
     }
 
+    /**
+     * Emits a signal when the router configuration has been loaded
+     *
+     * This signal can be used to add routes during runtime. In the slot make sure to receive the
+     * $routerConfiguration array by reference so you can alter it.
+     *
+     * @param array $routerConfiguration The router configuration
+     * @return void
+     * @Flow\Signal
+     */
+    protected function emitConfigurationLoaded(array &$routerConfiguration): void
+    {
+    }
+
+    public function configureRouteAnnotatedMethods(array &$routerConfiguration): void
+    {
+        $annotatedRoutes = static::resolveRouteAnnotatedMethods($this->objectManager);
+        $routerConfiguration = array_merge(array_values($annotatedRoutes), $routerConfiguration);
```

Comment:
That's a critical part. I don't think it's enough to just prepend configured routes with the ones from the annotations.

Example:
Imagine some 3rd party package with the following code (and no other routes configured at all):

```php
class SomeController {

  /**
   * @Flow\Route("some/path")
   **/
  public function someAction() {}

}
```

and now I'd like to change the path (without using AOP). So I add some `Routes.yaml` à la:

```yaml
-
  uriPattern: 'some/modified/path'
  defaults:
    '@package': 'Some.Package'
    '@controller': 'SomeController'
    '@action': 'someAction'
```

In theory it should be possible to end up with only _one_ effective route (the one from `Routes.yaml`) in this case. But at least it should be evaluated _before_  the annotation based routes

Reply:
Again, I would not consider this a good use-case. You should not annotate an action in a base package but only in your final application, as that route can then not be overwritten. Specifying the route for an action directly onto the method should be like a `final Route`, or `!important` inline style. If you read this, you know exactly how it will behave, no matter what else may exist somewhere else.

Yes I know that goes somewhat against the implicit Flow convention of having things changeable in global settings, but it would create a developer experience hell in this case.

Reply:
It's not a good use case I agree. But I can already see "base" packages that make use of this, e.g. Neos.Media – And IMO that's perfectly fine and we should not forbid that.
But I think that we should be able to _override_ the default behavior (e.g. change the action behind `/neos/media/some-service/foo`)

And this would be easily possible if we just evaluate the annotated routes _after_ the yaml-routes or am I missing something?

Reply:
Resolving this based on conclusion in last comment and after call